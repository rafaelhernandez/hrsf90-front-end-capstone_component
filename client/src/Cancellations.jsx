import React from 'react';

class Cancellations extends React.Component {
  constructor(props) {
    super(props);
  }

  cancellationExplanation(cancellationPolicy) {
    let explanation = '';
    if (cancellationPolicy === 'flexible') {
      explanation = 'Cancel up to 24 hours before check in and get a full refund (minus service fees). Cancel within 24 hours of your trip and the first night is non-refundable. Service fees are refunded when cancellation happens before check in and within 48 hours of booking.';
    }
    return explanation;
  }

  render() {
    return (
      <div className="div-description-cancellations">
        <div className="div-description-cancellations-type">
          {this.props.cancellationPolicy.charAt(0).toUpperCase() + this.props.cancellationPolicy.slice(1)}
        </div>
        {this.cancellationExplanation(this.props.cancellationPolicy)}
      </div>
    );
  }
}


export default Cancellations;
