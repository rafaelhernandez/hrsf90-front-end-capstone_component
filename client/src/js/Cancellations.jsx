import React from 'react';

class Cancellations extends React.Component {
  constructor(props) {
    super(props);
  }

  cancellationExplanation(cancellationPolicy) {
    let explanation = '';
    if (cancellationPolicy === 'flexible') {
      explanation = 'Cancel up to 24 hours before check in and get a full refund (minus service fees). Cancel within 24 hours of your trip and the first night is non-refundable. Service fees are refunded when cancellation happens before check in and within 48 hours of booking.';
    }
    return explanation;
  }

  render() {
    return (
      <div className="div-description-cancellations">
        <div className="div-description-section-header">
          Cancellations
        </div>
        <div className="div-description-cancellations-type div-description-separator-margin">
          {this.props.cancellationPolicy.charAt(0).toUpperCase() + this.props.cancellationPolicy.slice(1)}
        </div>
        {this.cancellationExplanation(this.props.cancellationPolicy)}
        <div>
          <p className="div-description-separator-margin">
            <a className="a-description-fold" href="#">Get details</a>
          </p>
        </div>
      </div>
    );
  }
}


export default Cancellations;
