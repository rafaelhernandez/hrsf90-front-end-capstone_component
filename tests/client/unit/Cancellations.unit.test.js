import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Cancellations from '../../../client/src/js/Cancellations.jsx';
import descriptionSample from '../../../db/data/description_sample.js';

configure({ adapter: new Adapter() });

describe('Cancellations', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Cancellations cancellationPolicy={descriptionSample.cancellation_policy}></Cancellations>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
