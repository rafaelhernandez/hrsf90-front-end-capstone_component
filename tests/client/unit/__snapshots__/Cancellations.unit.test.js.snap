// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cancellations should render correctly 1`] = `
<div
  className="div-description-cancellations"
>
  <div
    className="div-description-cancellations-type"
  >
    Flexible
  </div>
  Cancel up to 24 hours before check in and get a full refund (minus service fees). Cancel within 24 hours of your trip and the first night is non-refundable. Service fees are refunded when cancellation happens before check in and within 48 hours of booking.
</div>
`;
